src/mono/mono/metadata/appdomain.c

```diff
@@ -774,6 +774,12 @@ ves_icall_System_Reflection_Assembly_InternalLoad (MonoStringHandle name_handle,
 	MonoAssembly *requesting_assembly = mono_runtime_get_caller_from_stack_mark (stack_mark);
 	MonoAssemblyLoadContext *alc = (MonoAssemblyLoadContext *)load_Context;
 
+#if HOST_WASI
+	// On WASI, mono_assembly_get_alc isn't yet supported. However it should be possible to make it work.
+	if (!alc)
+		alc = mono_alc_get_default ();
+#endif
```

Comment:
@SteveSandersonMS Do you have a tracking issue for WASI support?  Might be good to start a task list of these kind of missing bits.

Reply:
Is there a stack trace for when this happens ? Maybe the embedding app doesn't setup something correctly ?

Reply:
Exactly. I was looking for it. Didn't find. The issue will be easier to track PRs in one place and the community can discuss and learn as well along with these changes!

Reply:
Great idea - I've filed https://github.com/dotnet/runtime/issues/65895. There will probably be a bunch more things to add there over time.

Reply:
> Is there a stack trace for when this happens ?

Yes:

```
Error: wasm trap: out of bounds memory access
wasm backtrace:
    0: 0xc851 - <unknown>!mono_image_get_alc
    1: 0xc80b - <unknown>!mono_assembly_get_alc
    2: 0xc481 - <unknown>!ves_icall_System_Reflection_Assembly_InternalLoad
    3: 0xa55ba - <unknown>!ves_icall_System_Reflection_Assembly_InternalLoad_raw
    4: 0x3e1df6 - <unknown>!do_icall
    5: 0x3de329 - <unknown>!do_icall_wrapper
    6: 0x389273 - <unknown>!interp_exec_method
    7: 0x385119 - <unknown>!interp_runtime_invoke
    8: 0x30b64a - <unknown>!mono_jit_runtime_invoke
    9: 0x12a6ad - <unknown>!do_runtime_invoke
   10: 0x12c396 - <unknown>!mono_runtime_try_invoke
   11: 0x135eb5 - <unknown>!mono_runtime_invoke
   12: 0x40ea84 - <unknown>!mono_wasm_invoke_method
   13: 0x379b - <unknown>!start_dotnet
   14: 0x37e1 - <unknown>!__original_main
   15: 0x3622 - <unknown>!_start
```

More info here though: https://github.com/dotnet/runtime/pull/65888#discussion_r814829756

Reply:
Yes @SteveSandersonMS, that's great. Maybe later you can put some links there about WASI stuff so that people can visit and learn more, as this is something totally new! 

Maybe this might be a TOC of issue. 
1. Heading as this is experimental 
2. WASI small intro in 2 lines. 
3. WASI official docs
4. List of efforts that the team (maybe you just at the moment) is planning/experimenting with. Link's to PR as well, if it's done!  
List of obstacles as of now (from .NET perspective, I heard one of them is ECMAScriptan that we are using as of now!) 
Sample projects with people can test out if they wanna!

Reply:
Will investigate that, stack walks should definitely work with the interpreter even on wasm.